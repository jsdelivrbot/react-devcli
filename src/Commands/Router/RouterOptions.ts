export const name = 'router';

export const description = 'router description';

export const flags = {};
