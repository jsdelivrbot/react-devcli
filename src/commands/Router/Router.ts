import {Arguments} from 'yargs';
import ACommand from '../Command/ACommand';
import {CommandsConfig} from 'react-devcli';
import {IMakeable, IRemovable} from '../Command/contracts/ISubCommands';

class Router extends ACommand implements IRemovable, IMakeable {
    public constructor() {
        const name = 'router';
        const description = 'router description';
        const options = {};

        super(name, description, options);
    }

    public run(argv: Arguments, config: CommandsConfig): void {
        console.log(config);
        console.log(argv);
    }

    public help(argv: Arguments): void {
        console.log(argv);
    }

    public remove(argv: Arguments): void {
        console.log(argv);
    }

    public make(argv: Arguments): void {
        console.log(argv);
    }
}

export default new Router();
