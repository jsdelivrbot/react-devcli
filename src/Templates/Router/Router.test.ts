import {js, ts} from './Router';

describe('Router template test suite', (): void => {
    test('should match router template snapshot for JS language', (): void => {
        expect(js()).toMatchSnapshot();
    });

    test('should match router template snapshot for TS language', (): void => {
        expect(ts()).toMatchSnapshot();
    });
});
