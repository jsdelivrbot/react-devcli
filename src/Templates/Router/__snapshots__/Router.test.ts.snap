// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router template test suite should match router template snapshot for JS language 1`] = `""`;

exports[`Router template test suite should match router template snapshot for TS language 1`] = `""`;
